chart/templates/_helpers.yaml

```diff
@@ -223,27 +223,28 @@ If release name contains chart name it will be used as a full name.
     - name: GITSYNC_SSH_KNOWN_HOSTS
       value: "false"
     {{- end }}
-    {{ else if .Values.dags.gitSync.credentialsSecret }}
-    - name: GIT_SYNC_USERNAME
+    {{- else if and .Values.dags.gitSync.credentialsSecret (semverCompare ">=v4.0.0" .Values.images.gitSync.tag)}}
```

Comment:
We can't rely on the tag to determine the version, unfortunately. Someone could easily mirror the image and change the tagging approach (e.g. `2024-05-08`).

If we need to do it conditionally, we need to introduce another version config like we have for `airflowVersion`.